### DISCUSSION TITLE
Layout is switched before data is loaded from server

### DISCUSSION THREAD

**Discussion by pavloniym** [nuxt/nuxt#19849] [Questions]
Hello to everyone!

### My dependencies:

* Nuxt v3.3.1
* Node v19
* Vuetify 3.11.1

### Problem:
Layout is switched before data is fetch from server

---
`app.vue`
```vue

<template>
    <v-app>
        <nuxt-layout>
            <nuxt-page />
        </nuxt-layout>
    </v-app>
</template>

```

`layouts/app.vue`

```vue

<template>
    <div>

        <!-- Header -->
        <app-header />

        <!-- Main -->
        <!-- Content -->
        <v-main class='h-100'>
            <v-container>
                <slot />
            </v-container>
        </v-main>

    </div>
</template>
```

`layouts/clean.vue`

```vue
<template>
    <div class='h-100vh'>
        <slot />
    </div>
</template>
```

I have custom `router.options.ts` in my `app` folder with custom routing files:

`homeRoutes.js`
```js
export const HOME_ROUTE = 'HOME_ROUTE'
export default [
    {
        path: '/',
        name: HOME_ROUTE,
        meta: { layout: 'app' },
        component: () => import('~/pages/home/HomePage.vue')
    }
]
```

`watchEpisodeRoutes.js`
```js
export const WATCH_EPISODE_ROUTE = 'WATCH_EPISODE_ROUTE'
export default [
    {
        path: '/watch/episode/:episodeId',
        name: WATCH_EPISODE_ROUTE,
        meta: { layout: 'clean' },
        props: true,
        component: () => import('~/pages/watch/episode/WatchEpisodePage.vue')
    }
]
```

As you can see - the layout is defined in my custom routes files (same as nuxt do it automatically in general page-creation-way)
I use `useFetch` to fetch content from server in my HomePage.vue.

`HomePage.vue`
```vue
<script setup lang='ts'>
    const latestReleases = await useFetch('/api/v1/anime/releases/latest', { query: {limit: 42}, method: 'GET', })
</script>
```

When I reload my page in browser (Cmd+R) — everything is ok, page with `app` layout is mounted after data is fetched!
But when I push router from `WatchEpisodePage.vue` (`clean` layout) to `HomePage.vue` (`app` layout) — nuxt starts to fetch data from server AND changes current page immediately without waiting of `useFetch` completion. 
So I get `app` layout with no data for some time, then data is loaded and added to page.

![chrome-capture-2023-2-22 (1)](https://user-images.githubusercontent.com/17234942/226752264-65034240-267c-492a-acb7-5757e4f1d757.gif)

As you can see layout (top bar is appeared) is switched before contend is loaded (json is appeared)
What am i doing wrong? Please, help!

**Comment by sawa-ko** [comment]
This PR should correct that problem of asynchronous components. (Or so I hope)
https://github.com/nuxt/nuxt/pull/19605

**Comment by pavloniym** [comment]
@sawa-ko 
Same behaviour with nuxt [v3.3.2](https://github.com/nuxt/nuxt/releases/tag/v3.3.2)

**Comment by sawa-ko** [comment]
Yes, unfortunately it is still the same. Any idea @danielroe?
